src/move/emojicoin_dot_fun/sources/emojicoin_dot_fun.move

```diff
@@ -188,6 +249,95 @@ module emojicoin_dot_fun::emojicoin_dot_fun {
         )
     }
 
+    inline fun get_registry_address(): address {
+        borrow_global<RegistryAddress>(@emojicoin_dot_fun).registry_address
+    }
+
+    inline fun get_market_address(emoji_bytes: vector<u8>): Option<address> {
+        let registry = borrow_global<Registry>(get_registry_address());
+        let markets = &registry.markets_by_emoji_bytes;
+        if (smart_table::contains(markets, emoji_bytes)) {
+            option::some(*smart_table::borrow(markets, emoji_bytes))
+        } else {
+            option::none()
+        }
+    }
```

Comment:
Thoughts on doing this with an address return and an abort if market doesn't exist?

Will require less conditional branching and func calls to option module

Reply:
Can follow up in future PR

Currently prioritizing merge to prevent bloat, since other content needs to merge too